modules/dists/BlockDist.chpl

```diff
@@ -1657,6 +1657,94 @@ where useBulkTransferDist {
   return true;
 }
 
+config param debugBlockScan = false;
+
+proc BlockArr.doiScan(op, dom) where (rank == 1) &&
+                                     chpl__scanStateResTypesMatch(op) {
+
+  // The result of this scan, which will be Block-distributed as well
+  type resType = op.generate().type;
+  var res: [dom] resType;
+
+  // Store one element per locale in order to track our local total
+  // for a cross-locale scan as well as flags to negotiate reading and
+  // writing it.  This domain really wants an easier way to express
+  // it...
+  use ReplicatedDist;
+  ref targetLocs = this.dsiTargetLocales();
+  const elemPerLocDom = {1..1} dmapped Replicated(targetLocs);
+  var elemPerLoc: [elemPerLocDom] resType;
+  var inputReady$: [elemPerLocDom] sync bool;
```

Comment:
Using atomics (particularly when network atomics are available) will likely be much faster, but that's probably not important for the first version.

Reply:
I thought about that, but it's really the ideal case for sync vars—one task can't proceed until the others have written their results; and then the others can't continue until the one task finishes updating their value.

Reply:
(so I had a reaction like "if not here, then where?")